editor/src/components/canvas/ui-jsx-canvas-renderer/ui-jsx-canvas-execution-scope.tsx

```diff
@@ -143,3 +157,36 @@ export function createExecutionScope(
     requireResult: requireResult,
   }
 }
+
+export function useGetCodeAndHighlightBounds(
+  filePath: string | null,
+): { code: string; highlightBounds: HighlightBoundsForUids | null } {
+  return useContextSelector(UtopiaProjectContext, (c) => {
```

Comment:
beware, `useContextSelector`'s default equality function is `Object.is` (strict equality) and not shallowEq. you can change that by providing a shallow eq function as the third parameter